src/backend/aspen/api/views/phylo_runs.py

```diff
@@ -193,25 +193,42 @@ async def kick_off_phylo_run(
     return PhyloRunResponse.from_orm(workflow)
 
 
-async def get_editable_phylo_run_by_id(db, run_id, user):
+async def get_editable_phylo_runs_by_id(db, user, run_id=None):
+    # get list of all viewable phylo_runs viewable to user, otherwise returh phylo_run matching run_id if sufficient permissions
+    cansee_owner_group_ids: Set[int] = {
+        cansee.owner_group_id
+        for cansee in user.group.can_see
+        if cansee.data_type == DataType.TREES
+    }
     query = (
         sa.select(PhyloRun)
+        .options(
+            joinedload(PhyloRun.outputs.of_type(PhyloTree)),
+            joinedload(PhyloRun.user),  # For Pydantic serialization
+            joinedload(PhyloRun.group),  # For Pydantic serialization
+        )
         .filter(
-            sa.and_(
-                PhyloRun.group == user.group,  # This is an access control check!
-                PhyloRun.id == run_id,
-            )
+            sa.or_(
+                PhyloRun.group == user.group,
+                user.system_admin,
```

Comment:
This function is `get_editable_phylo_runs_by_id` and system admins aren't currently allowed to edit/delete phylo runs for groups other than the one they belong to. I'm not sure if we need a separate `get_readable_phylo_runs` function, or to be able to switch between queries for editable vs readable runs.

Reply:
i updated it within this one function